### ISSUE TITLE
Parallel state should allow Catch property

### ISSUE THREAD

**Issue by mrgrain** [airware/asl-validator#2]
Given a Parallel state
```
{
  "States": {
    "A": {
      "Type": "Parallel",
      "Branches": [
        {
          "StartAt": "B",
          "States": {
            "B": {
              "Type": "Fail"
            }
          }
        },
        {
          "StartAt": "C",
          "States": {
            "C": {
              "Type": "Pass",
              "End": true
            }
          }
        }
      ],
      "Catch": {
        "ErrorEquals": [
          "States.ALL"
        ],
        "Next": "D"
      }
    },
    "D": {
      "Type": "Fail"
    }
  }
}
````
It should allow the `Catch` property: http://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-parallel-state.html?shortFooter=true

However it fails. 

Version 1.0.2

**Comment by ChristopheBougere** [comment]
Hi @mrgrain,
You are right, it is missing the `Catch` property in `Parallel`.
However, in your example:
- It is missing the `StartAt` property of the state machine, which makes it fail at first
- You must provide either `Next`or `End` in your `Parallel` state
- `Catch` property must be an array, [as defined here](http://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-errors.html#amazon-states-language-fallback-states)

Which results in this:
```json
{
  "StartAt": "A",
  "States": {
    "A": {
      "Type": "Parallel",
      "Next": "D",
      "Branches": [{
          "StartAt": "B",
          "States": {
            "B": {
              "Type": "Fail"
            }
          }
        },
        {
          "StartAt": "C",
          "States": {
            "C": {
              "Type": "Pass",
              "End": true
            }
          }
        }
      ],
      "Catch": [{
        "ErrorEquals": [
          "States.ALL"
        ],
        "Next": "D"
      }]
    },
    "D": {
      "Type": "Fail"
    }
  }
}
```
Fixed in 1.0.3. Feel free to reopen if needed

**Comment by mrgrain** [comment]
Thanks, in the example I was trying to simplify my state machine - obviously not with much success.
